it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -921,9 +921,108 @@ class PersistenceConnectorSpec
     }
   }
 
-  private def generateDepartureResponse(departureId: MovementId) =
-    DepartureResponse(
-      _id = departureId,
+  "GET /traders/:EORI/movements/arrivals" - {
+
+    implicit val hc = HeaderCarrier()
+    val eori        = arbitrary[EORINumber].sample.get
+
+    def targetUrl(eoriNumber: EORINumber) = s"/transit-movements/traders/${eoriNumber.value}/movements/arrivals"
+
+    "on success, return a list of arrival IDs" in {
+      lazy val arrivalIdList = Gen.listOfN(3, arbitrary[MovementId]).sample.get
+      lazy val messageResponse = arrivalIdList.map(
+        id => generateMovementResponse(id)
+      )
+
+      server.stubFor(
+        get(
+          urlEqualTo(targetUrl(eori))
+        )
+          .willReturn(
+            aResponse()
+              .withStatus(OK)
+              .withBody(
+                Json.toJson(messageResponse).toString()
+              )
+          )
+      )
+
+      implicit val hc = HeaderCarrier()
+      val result      = persistenceConnector.getArrivalsForEori(eori)
+      whenReady(result) {
+        _ mustBe messageResponse
+      }
+    }
+
+    "on incorrect Json, return an error" in {
+      server.stubFor(
+        get(
+          urlEqualTo(targetUrl(eori))
+        )
+          .willReturn(
+            aResponse()
+              .withStatus(OK)
+              .withBody(
+                "{ \"test\": \"fail\" }"
+              )
+          )
+      )
+
+      val result = persistenceConnector
+        .getArrivalsForEori(eori)
+        .map(
+          _ => fail("This should have failed with a JsResult.Exception, but it succeeded")
+        )
+        .recover {
+          case JsResult.Exception(_)  => ()
+          case t: TestFailedException => t
+          case thr                    => fail(s"Expected a JsResult.Exception, got $thr")
+        }
+
+      whenReady(result) {
+        _ mustBe (())
+      }
+    }
+
+    "on an internal error, return an UpstreamServerError" in {
+      server.stubFor(
+        get(
+          urlEqualTo(targetUrl(eori))
+        )
+          .willReturn(
+            aResponse()
+              .withStatus(INTERNAL_SERVER_ERROR)
+              .withBody(
+                Json.stringify(
+                  Json.obj(
+                    "code"    -> "INTERNAL_SERVER_ERROR",
+                    "message" -> "Internal server error"
+                  )
+                )
+              )
+          )
+      )
+
+      val result = persistenceConnector
+        .getArrivalsForEori(eori)
+        .map(
+          _ => fail("This should have failed with an UpstreamErrorResponse, but it succeeded")
```

Comment:
This working could be improved . Something like below

          .map(
            _ => fail("This should not succeed")
          )
          .recover {
            case UpstreamErrorResponse(_, INTERNAL_SERVER_ERROR, _, _) =>
          }

Reply:
The wording is perfectly fine -- in fact, what has been written here is better because it explains exactly what happened.

Reply:
No problem. but this test case look like is written to cover 500 Internal server error. so how it can succeed?  the wording says but it succeeded. Still I am confused.

Reply:
The `map` is only executed when the call succeeds. Since we're checking that we handle the _failure_ case correctly, then if we end up in the `map` method, we didn't get an error back and so the call succeeded -- so this should fail the test.

Reply:
Ok fine. I misunderstood this.